### ISSUE TITLE
pillar::squeeze() is slow, and causes slow printing of tibbles

### ISSUE THREAD

**Issue by kevinushey** [r-lib/pillar#139]
This was a bit surprising to me:

```R
> library(tibble)
> system.time(print(as_data_frame(mtcars)))
# A tibble: 32 x 11
     mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
   <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
 2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
 3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
 4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
 5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
 6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
 7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
 8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
 9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
# … with 22 more rows
   user  system elapsed
  0.607   0.013   0.624
```

profivs suggests all the time is spent in `pillar::squeeze()` / `pillar_shaft()`. Performance here isn't vital but 0.6s is just long enough to be surprising.

<details>
<summary>Session Info</summary>

```r
> devtools::session_info()
Session info ------------------------------------------------------------------
 setting  value
 version  R version 3.5.1 (2018-07-02)
 system   x86_64, darwin15.6.0
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 tz       America/Los_Angeles
 date     2018-10-07

Packages ----------------------------------------------------------------------
 package    * version     date       source
 assertthat   0.2.0       2017-04-11 CRAN (R 3.5.0)
 base       * 3.5.1       2018-07-05 local
 cli          1.0.1       2018-09-25 CRAN (R 3.5.0)
 commonmark   1.6         2018-09-30 CRAN (R 3.5.0)
 compiler     3.5.1       2018-07-05 local
 crayon       1.3.4       2017-09-16 CRAN (R 3.5.0)
 datasets   * 3.5.1       2018-07-05 local
 devtools   * 1.13.6      2018-06-27 CRAN (R 3.5.0)
 digest       0.6.17      2018-09-12 CRAN (R 3.5.1)
 evaluate     0.11        2018-07-17 CRAN (R 3.5.1)
 fansi        0.4.0       2018-10-05 CRAN (R 3.5.0)
 graphics   * 3.5.1       2018-07-05 local
 grDevices  * 3.5.1       2018-07-05 local
 htmldeps     0.1.1       2018-09-13 Github (rstudio/htmldeps@c1023e0)
 htmltools    0.3.6       2017-04-28 CRAN (R 3.5.0)
 knitr      * 1.20.15     2018-09-13 Github (yihui/knitr@4cf7a6f)
 magrittr     1.5         2014-11-22 CRAN (R 3.5.0)
 memoise      1.1.0       2017-04-21 CRAN (R 3.5.0)
 methods    * 3.5.1       2018-07-05 local
 pillar       1.3.0.9000  2018-09-02 Github (r-lib/pillar@3fabb4e)
 pkgconfig    2.0.2       2018-08-16 CRAN (R 3.5.1)
 R6           2.3.0       2018-10-04 CRAN (R 3.5.0)
 Rcpp         0.12.19     2018-10-01 CRAN (R 3.5.0)
 rlang        0.2.2.9001  2018-09-18 Github (r-lib/rlang@4d697ca)
 rmarkdown  * 1.10.13     2018-09-18 local
 roxygen2   * 6.1.0       2018-07-27 CRAN (R 3.5.0)
 stats      * 3.5.1       2018-07-05 local
 stringi      1.2.4       2018-07-20 CRAN (R 3.5.1)
 stringr      1.3.1       2018-05-10 CRAN (R 3.5.0)
 testthat   * 2.0.0       2017-12-13 CRAN (R 3.5.0)
 tibble     * 1.4.99.9004 2018-09-02 Github (tidyverse/tibble@e54aeeb)
 tools        3.5.1       2018-07-05 local
 utf8         1.1.4       2018-05-24 CRAN (R 3.5.0)
 utils      * 3.5.1       2018-07-05 local
 withr        2.1.2       2018-03-15 CRAN (R 3.5.0)
 xfun         0.3         2018-07-06 CRAN (R 3.5.0)
 xml2         1.2.0       2018-01-24 CRAN (R 3.5.0)
>
```

</details>

**Comment by krlmlr** [comment]
Thanks for the nudge.

**Comment by krlmlr** [comment]
Closed in 036f338.

**Comment by kevinushey** [comment]
Much faster now :-)

``` r
> library(tibble)
> system.time(print(as_data_frame(mtcars)))
# A tibble: 32 x 11
     mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
 * <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
 2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
 3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
 4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
 5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
 6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
 7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
 8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
 9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
# ... with 22 more rows
   user  system elapsed
  0.048   0.006   0.060
```

Thanks!
